include/deal.II/lac/block_linear_operator.h

```diff
@@ -325,98 +325,104 @@ block_diagonal_operator(const LinearOperator<typename Range::BlockType, typename
 /**
  * @relates LinearOperator
  *
- * A function that takes an array of arrays of LinearOperators @p block_matrix
- * and returns its associated lower triangular matrix operator
- * (diagonal is not included).
+ * This function implement a forward substitution argument to invert a lower
+ * block triangular matrix.
+ * It takes as argement an array of array of LinearOperators @p block_matrix
+ * representing a block matrix and an array of LinearOperators
+ * @p inverse_diagonal representing inverses of digonal blocks of @p block_matrix.
  *
- * @code
- *  a00 | a01 | a02                 |     |
- *  ---------------             ---------------
- *  a10 | a11 | a12     ->      a10 |     |
- *  ---------------             ---------------
- *  a20 | a21 | a22             a20 | a21 |
- * @endcode
+ * Let us assume we have a linear system where each coefficient of the system is a
+ * matrix:
+ * A00 x0 + ...                   = y0
+ * A01 x0 + A11 x1 + ...          = y1
+ * ...        ...
+ * A0n x0 + A1n x1 + ... + Ann xn = yn
  *
+ * First of all, x0 = A00^-1 y0
+ * Then, we can use x0 to recover x1:
+ *    x1 = A11^-1 ( y1 - A01 x0 )
+ * and therefore:
+ *    xn = Ann^-1 ( yn - A0n x0 - ... - A(n-1)n x(n-1) )
+ *
+ * Caveat: Tvmult and Tvmult_add have not been implemented, yet. This may lead to mistakes.
```

Comment:
I'd remove the phrase "This may lead to mistake", since the default code of LinearOperator is to throw an exception if called when you do not implement Tvmult...